doc/source/cross-language.rst

```diff
@@ -0,0 +1,246 @@
+Cross Language
+==============
+
+This page will cover some more cross language examples of using Ray's flexible programming model.
+
+Python calling Java
+-------------------
+
+You can write a Java class as follows:
+
+.. code-block:: java
+
+  // A regular Java class.
+  public class Counter {
+
+    private int value = 0;
+
+    public int increment() {
+      this.value += 1;
+      return this.value;
+    }
+
+    public static int add(int a, int b) {
```

Comment:
Maybe put this in a different class. E.g., `Math.add`. It's a bit weird to put it here.

Reply:
OK. I will add a new class for the `add` function.